### ISSUE TITLE
ionic 4 Beta: Package empty

### ISSUE THREAD

**Issue by Turkzilla** [sneas/ionic-native-http-connection-backend#59]
installing ionic-native-http-connection-backend@last brings an empty package. Installing regular ("^4.0.7"), will not compile: 
ERROR in node_modules/ionic-native-http-connection-backend/dist/http/native-http-backend.d.ts(11,43): error TS2304: Cannot find name 'HTTP'. (Yes, @ionic-native/http installed.)

```
Ionic:

   ionic (Ionic CLI)             : 4.2.1 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.0.0-beta.13
   @angular-devkit/build-angular : 0.9.0-beta.3
   @angular-devkit/schematics    : 0.9.0-beta.3
   @angular/cli                  : 6.2.5
   @ionic/angular-toolkit        : 1.0.0

Cordova:

   cordova (Cordova CLI) : 8.1.2 (cordova-lib@8.1.1)
```

```
package.json:
"dependencies": {
    "@angular/common": "~6.1.1",
    "@angular/core": "~6.1.1",
    "@angular/forms": "~6.1.1",
    "@angular/http": "~6.1.1",
    "@angular/platform-browser": "~6.1.1",
    "@angular/platform-browser-dynamic": "~6.1.1",
    "@angular/pwa": "^0.8.4",
    "@angular/router": "~6.1.1",
    "@angular/service-worker": "~6.1.1",
    "@ionic-native/core": "5.0.0-beta.21",
    "@ionic-native/http": "^5.0.0-beta.21",
    "@ionic-native/native-storage": "^4.15.0",
    "@ionic-native/splash-screen": "5.0.0-beta.21",
    "@ionic-native/status-bar": "5.0.0-beta.21",
    "@ionic-native/toast": "^4.15.0",
    "@ionic/angular": "^4.0.0-beta.13",
    "@ionic/storage": "^2.2.0",
    "cordova-android": "7.1.1",
    "cordova-browser": "5.0.4",
    "cordova-ios": "^4.5.5",
    "cordova-plugin-advanced-http": "^2.0.1",
    "cordova-plugin-certificates": "^0.6.4",
    "cordova-plugin-device": "^2.0.2",
    "cordova-plugin-file": "^6.0.1",
    "cordova-plugin-ionic-keyboard": "^2.1.3",
    "cordova-plugin-ionic-webview": "^2.2.0",
    "cordova-plugin-nativestorage": "^2.3.2",
    "cordova-plugin-splashscreen": "^5.0.2",
    "cordova-plugin-statusbar": "^2.4.2",
    "cordova-plugin-whitelist": "^1.3.3",
    "cordova-plugin-x-toast": "^2.7.0",
    "core-js": "^2.5.3",
    "ionic-native-http-connection-backend": "^4.0.7",
    "moment": "^2.22.2",
    "moment-timezone": "^0.5.21",
    "rxjs": "^6.3.3",
    "rxjs-compat": "^6.3.3",
    "zone.js": "^0.8.26"
  },
  "devDependencies": {
    "@angular-devkit/architect": "^0.9.0-beta.3",
    "@angular-devkit/build-angular": "^0.9.0-beta.3",
    "@angular-devkit/core": "^0.9.0-beta.3",
    "@angular-devkit/schematics": "^0.9.0-beta.3",
    "@angular/cli": "^6.2.4",
    "@angular/compiler": "~6.1.1",
    "@angular/compiler-cli": "~6.1.1",
    "@angular/language-service": "~6.1.1",
    "@ionic/angular-toolkit": "^1.0.0",
    "@ionic/lab": "1.0.11",
    "@ionic/ng-toolkit": "^1.1.0",
    "@ionic/schematics-angular": "^1.0.0",
```

**Comment by sneas** [comment]
You got to follow the HTTP plugin installation steps to make it work https://ionicframework.com/docs/native/http/#installation

**Comment by sneas** [comment]
Hmmm, this is unexpectedly... I don’t have access to computer until 27th of October. Will have a look at the issue as soon as I get back home.

**Comment by Turkzilla** [comment]
Copying the src and using directly worked perfectly! Thank you, you have my gratitude for this awesome work!

The app totally switched over to native http on devices and used the httpclient on browser. I also made a minor insert to your code to make the app by-pass SSL check ( self-signed certificate ) on devices during testing:

**src/native-http-backend.ts**:
```

            this.nativeHttp.setDataSerializer(
                this.detectDataSerializerType(req),
            );
            this.nativeHttp.setSSLCertMode('nocheck'); // REMOVE FOR PRODUCTION
            this.nativeHttp[requestMethod](url, body, { ...headers })
                .then((response: HTTPResponse) => {
                    fireResponse({
                        body: response.data,
                        status: response.status,
                        headers: response.headers,
                    });
                })
```

**Comment by sneas** [comment]
Fixed with 39d497123dc804b8f78376d2e36142176392db5c

Version 5.0.1 created.
